src/Tribe/Aggregator/Service.php

```diff
@@ -0,0 +1,255 @@
+	 *
+	 * @return stdClass
+	 */
+	public function get( $endpoint, $data = array() ) {
+		$url = $this->build_url( $endpoint );
+		$url = esc_url_raw( add_query_arg( $data, $url ) );
+
+		$response = wp_remote_get( $url );
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		// if the response is not an image, let's json decode the body
+		if ( ! preg_match( '/image/', $response['headers']['content-type'] ) ) {
+			$response = json_decode( wp_remote_retrieve_body( $response ) );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Performs a POST request against the Event Aggregator service
+	 *
+	 * @param string $url Endpoint URL
+	 * @param array $data Array of parameters to send to the endpoint
+	 *
+	 * @return stdClass
+	 */
+	public function post( $endpoint, $data = array() ) {
+		$url = $this->build_url( $endpoint );
+
+		if ( empty( $data['body'] ) ) {
+			$args = array( 'body' => $data );
+		} else {
+			$args = $data;
+		}
+
+		$response = wp_remote_post( $url, $args );
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$response = json_decode( wp_remote_retrieve_body( $response ) );
+
+		return $response;
+	}
+
+	/**
+	 * Fetch origins from service
+	 *
+	 * @return array
+	 */
+	public function get_origins() {
+		$origins = array(
+			(object) array(
+				'id' => 'csv',
+				'name' => __( 'CSV File', 'the-events-calendar' ),
+			),
+		);
+
+		// if the user doesn't have a license key, don't bother hitting the service
+		if ( ! $this->api_key ) {
+			return $origins;
+		}
+
+		$response = $this->get( 'origin' );
+
+		if ( $response && 'success' === $response->status ) {
+			$origins = array_merge( $origins, $response->data->origin );
+		}
+
+		return $origins;
+	}
+
+	/**
+	 * Fetch import data from service
+	 *
+	 * @return array
+	 */
+	public function get_import( $import_id ) {
+		// if the user doesn't have a license key, don't bother hitting the service
+		if ( ! $this->api_key ) {
+			return new WP_Error( 'invalid-ea-license', __( 'You must enter an Event Aggregator license key in Events > Settings > Licenses', 'the-events-calendar' ) );
+		}
+
+		$response = $this->get( 'import/' . $import_id );
+
+		return $response;
+	}
+
+	/**
+	 * Creates an import
+	 *
+	 * @param array $args {
+	 *     Array of arguments. See REST docs for details. 1 exception listed below:
+	 *
+	 *     @type array $source_file Source file array using the $_FILES array values
+	 * }
+	 *
+	 * @return string
+	 */
+	public function post_import( $args ) {
+		// if the user doesn't have a license key, don't bother hitting the service
+		if ( ! $this->api_key ) {
+			return new WP_Error( 'invalid-ea-license', __( 'You must enter an Event Aggregator license key in Events > Settings > Licenses', 'the-events-calendar' ) );
+		}
+
+		$request_args = array(
+			'body' => $args,
+		);
+
+		if ( isset( $args['source_file'] ) ) {
+			$boundary = wp_generate_password( 24 );
```

Comment:
It's not much of big problem to regenerate it, really inexpensive in terms of performance, but it could be a problem in terms of trying to split this string again later, an action that I can't see a reason to use 😁